backend/api/prisonRegisterApi.test.ts

```diff
@@ -0,0 +1,39 @@
+import nock from 'nock'
+
+import Client from './oauthEnabledClient'
+import PrisonRegisterApi from './prisonRegisterApi'
+
+const baseUrl = 'http://localhost:8080'
+
+describe('prisonRegisterApi', () => {
+  const api = new PrisonRegisterApi(new Client({ baseUrl, timeout: 2000 }))
+  const mock = nock(baseUrl)
+
+  afterEach(() => {
+    nock.cleanAll()
+  })
+
+  describe('getVideoLinkConferencingCentreEmailAddress', () => {
+    it('Gets a VCC email address', async () => {
+      mock.get('/secure/prisons/id/MDI/videolink-conferencing-centre/email-address').reply(200, 'vcc@def')
+      expect(await api.getVideoLinkConferencingCentreEmailAddress({}, 'MDI')).toEqual('vcc@def')
+    })
+
+    it('VCC email address not found', async () => {
+      mock.get('/secure/prisons/id/MDI/videolink-conferencing-centre/email-address').reply(404, {})
+      await expect(api.getVideoLinkConferencingCentreEmailAddress({}, 'MDI')).rejects.toThrow(new Error('Not Found'))
+    })
+  })
+
+  describe('getOffenderManagementUnitEmailAddress', () => {
+    it('Gets an OMU email address', async () => {
+      mock.get('/secure/prisons/id/MDI/offender-management-unit/email-address').reply(200, 'omu@def')
+      expect(await api.getOffenderManagementUnitEmailAddress({}, 'MDI')).toEqual('omu@def')
+    })
+
+    it('OMU email address not found', async () => {
+      mock.get('/secure/prisons/id/MDI/offender-management-unit/email-address').reply(404, {})
+      await expect(api.getOffenderManagementUnitEmailAddress({}, 'MDI')).rejects.toThrow(new Error('Not Found'))
```

Comment:
do we handle the 404s if these emails don't exist? 

If you look at secrets in dev, then you'll see that we occasionally have empty strings for some recipients that don't exist. The service just currently ignores them and doesn't send emails.
 
Will this error in that case?

Reply:
Hmm.  The client will reject the Promise as demonstrated in the test above. The NotificationService emailToRecipient method wraps everything in a try/catch block which will log the error and return a resolved Promise.

Reply:
Like the idea of caching, little bit conscious that we might be logging errors but happy to check this out in dev